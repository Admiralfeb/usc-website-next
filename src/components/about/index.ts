export * from './allies';
export * from './faction';
export * from './hc';
export * from './rules';
export * from './carriers';
