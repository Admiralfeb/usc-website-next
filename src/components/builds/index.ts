export * from './builds/buildAdd';
export * from './builds/buildDetail';
export * from './queryandBuildList';
