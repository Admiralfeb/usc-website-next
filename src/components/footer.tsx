import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';

function ImgCopyright() {
  return (
    <Typography
      variant='body2'
      color='textSecondary'
      style={{ textAlign: 'center' }}>
      Thank you to CMDR RaZ uryel for the background image
    </Typography>
  );
}
function Copyright() {
  return (
    <Typography
      variant='body2'
      color='textSecondary'
      style={{ textAlign: 'center' }}>
      {'Copyright Â© '}
      <Link color='inherit' href='https://unitedsystems.org/'>
        United Systems Cooperative
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor:
      theme.palette.type === 'light'
        ? theme.palette.grey[200]
        : theme.palette.grey[800],
  },
  center: {
    textAlign: 'center',
  },
}));

export const Footer = () => {
  const classes = useStyles();

  return (
    <footer className={classes.footer}>
      <Container maxWidth='sm' className={classes.center}>
        <Copyright />
        <ImgCopyright />
      </Container>
    </footer>
  );
};
