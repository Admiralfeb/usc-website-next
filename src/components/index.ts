export * from './home';
export * from './footer';
export * from './notFound';
export * from './auth/login';
