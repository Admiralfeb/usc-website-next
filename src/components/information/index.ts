export * from './infoSection';
export * from './infographic';
export * from './information';
export * from './miningMap';
export * from './shipReviews';
