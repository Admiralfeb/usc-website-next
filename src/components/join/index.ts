export * from './join';
export * from './joinList';
export * from './joinNew';
export * from './joinFormMember';
