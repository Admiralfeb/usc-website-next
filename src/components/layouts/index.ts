export * from './about';
export * from './primary';
