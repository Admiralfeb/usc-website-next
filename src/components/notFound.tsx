import {
  Container,
  Link,
  makeStyles,
  Paper,
  Typography,
} from '@material-ui/core';
import { NavLink } from 'react-router-dom';
import errorImg from 'assets/404.jpg';

const useStyles = makeStyles({
  root: {
    textAlign: 'center',
  },
  img: {
    maxWidth: '100%',
    height: 'auto',
  },
});

export const NotFound = () => {
  const classes = useStyles();
  return (
    <Container maxWidth="lg" className={classes.root}>
      <Paper>
        <Typography variant="h3">Mistakes were made</Typography>
        <Typography variant="subtitle1">
          You were hyperdicted... escape destruction by going{' '}
          <Link to="/home" component={NavLink}>
            home
          </Link>
          .
        </Typography>
        <img className={classes.img} alt="404 Error" src={errorImg} />
      </Paper>
    </Container>
  );
};
