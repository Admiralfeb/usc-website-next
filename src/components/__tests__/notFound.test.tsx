import { cleanup, render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { NotFound } from 'components/notFound';

let componentBody: RenderResult;

describe('Join', () => {
  beforeEach(() => {
    componentBody = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    );
  });

  afterEach(cleanup);

  it(`should match the snapshot`, () => {
    const { baseElement } = componentBody;
    expect(baseElement).toMatchSnapshot();
  });
});
