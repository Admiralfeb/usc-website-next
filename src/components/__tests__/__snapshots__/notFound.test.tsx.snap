// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join should match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root makeStyles-root-1 MuiContainer-maxWidthLg"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3"
        >
          Mistakes were made
        </h3>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1"
        >
          You were hyperdicted... escape destruction by going
           
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="/home"
          >
            home
          </a>
          .
        </h6>
        <img
          alt="404 Error"
          class="makeStyles-img-2"
          src="404.jpg"
        />
      </div>
    </div>
  </div>
</body>
`;
